sycl/doc/CompilerAndRuntimeDesign.md

```diff
@@ -434,6 +434,52 @@ unit)
 * `per_kernel` - enables emitting a separate module for each kernel
 * `off` - disables device code split
 
+#### CUDA support
+
+The driver supports compilation to CUDA when the `nvptx64-nvidia-cuda-sycldevice` is passed to `-fsycl-targets`.
+
+Unlike other AOT targets, intermediate objects never go through SPIR-V. Instead all separate compilation objects compile down to llvm bitcode. All produced bitcode depends on two libraries, `libdevice.bc` (provided by the CUDA SDK) and `libspirv-nvptx64--nvidiacl.bc` (built by the libclc project).
+
+During the device linking step (device linker box in the [Separate Compilation and Linking](#separate-compilation-and-linking) illustration), llvm bitcode objects for the CUDA target are linked together alongside `libspirv-nvptx64--nvidiacl.bc` and `libdevice.bc`, compiled to PTX using the NVPTX backend, and assembled into a cubin using the `ptxas` tool (part of the CUDA SDK). The PTX file and cubin are assembled together using `fatbinary` to produce a CUDA fatbin. The CUDA fatbin is then passed to the offload wrapper tool.
```

Comment:
Please add a TODO to update the picture. It needs to reflect current state rather than (or in addition to) the generic picture, and include PTX, SPIRV and AOT BEs as well as new tools and data items produced by post-link. I will take an AR to update this altogether.

Reply:
I'll add this. Might be good to add an editable version of it as well (there is only the png version for now).

Reply:
Yes, editable version is needed. I'll try to craft one